haystack/components/generators/chat/hugging_face_api.py

```diff
@@ -407,9 +410,21 @@ def _run_non_streaming(
 
         if hfapi_tool_calls := choice.message.tool_calls:
             for hfapi_tc in hfapi_tool_calls:
-                tool_call = ToolCall(
-                    tool_name=hfapi_tc.function.name, arguments=hfapi_tc.function.arguments, id=hfapi_tc.id
-                )
+                arguments = hfapi_tc.function.arguments
```

Comment:
I would extract this logic (which is used in two different places) into a small static method.
Something like
```python
@staticmethod
def _convert_hfapi_tool_call(hfapi_tool_call) -> Optional[ToolCall]:
  ...
```

Once you do so, you can easily add a unit test to test it.